packit_service/worker/checker/run_condition.py

```diff
@@ -0,0 +1,192 @@
+                url=self.data.event_dict.get("distgit_project_url")
+            )
+            git_ref = self.data.tag_name
+            version = self.data.event_dict.get("version")
+        elif self.data.event_type in (
+            github.issue.Comment.event_type(),
+            gitlab.issue.Comment.event_type(),
+        ):
+            project = self.service_config.get_project(
+                url=self.data.event_dict.get("dist_git_project_url")
+            )
+            git_ref = "HEAD"
+        elif self.data.event_type in (koji.tag.Build.event_type(),):
+            git_ref = "HEAD"
+            version = self.data.event_dict.get("version")
+        elif self.data.event_type in (koji.result.Build.event_type(),):
+            git_ref = self.data.event_dict.get("commit_sha")
+            version = self.data.event_dict.get("version")
+
+        extra_env = {}
+
+        if self.job_config.clone_repos_in_run_condition:
+            actions_handler = self.packit_api.up.actions_handler
+            if isinstance(project, PagureProject):
+                self.packit_api.dg.local_project.checkout_release(git_ref)
+                extra_env["PACKIT_DOWNSTREAM_REPO"] = str(
+                    self.packit_api.dg.local_project.working_dir
+                )
+                if version is None:
+                    version = self.packit_api.dg.specfile.expanded_version
+            else:
+                self.packit_api.up.local_project.checkout_release(git_ref)
+                extra_env["PACKIT_UPSTREAM_REPO"] = str(
+                    self.packit_api.up.local_project.working_dir
+                )
+                if version is None:
+                    version = self.packit_api.up.get_current_version()
+        else:
+            actions_handler = self.actions_handler
+            specfile_path = (
+                f"{self.job_config.downstream_package_name}.spec"
+                if isinstance(project, PagureProject)
+                else self.job_config.specfile_path
+            )
+            if version is None:
+                try:
+                    specfile_content = project.get_file_content(path=specfile_path, ref=git_ref)
+                except FileNotFoundError:
+                    pass
+                else:
+                    with Specfile(
+                        content=specfile_content, sourcedir=".", force_parse=True
+                    ) as specfile:
+                        version = specfile.expanded_version
+        try:
+            from sandcastle.exceptions import SandcastleCommandFailed
+
+            if not actions_handler.has_action(ActionName.run_condition):
+                return True
+            try:
+                actions_handler.run_action(
+                    actions=ActionName.run_condition, env=self.common_env(version, extra_env)
+                )
+            except (PackitCommandFailedError, SandcastleCommandFailed):
+                return False
```

Comment:
oof, I hope this doesn't trip false positives from network flakes, cluster, etc.

Reply:
It could. `SandcastleCommandFailed` has a `rc` attribute, but that still doesn't have to be the actual exit code of the command. Maybe it would be safer to expect a specific stdout output from the action?

Reply:
I guess it would be way too much to expect from them to provide a normal type instead of raw JSON 🤦